### ISSUE TITLE
Evaluating  "Learning Sparse Nonparametric DAGs"

### ISSUE THREAD

**Issue by tangyong** [quantumblacklabs/causalnex#34]
## Description

[Update]DAGs with NO TEARS
https://github.com/xunzheng/notears

Zheng, X., Dan, C., Aragam, B., Ravikumar, P., & Xing, E. P. (2020). Learning sparse nonparametric DAGs (AISTATS 2020, to appear).

https://arxiv.org/pdf/1909.13189.pdf

## Possible Implementation
https://github.com/xunzheng/notears

pl. evaluating 1909.13189

Thanks!

**Comment by qbphilip** [comment]
Hello Tang Yong,

Thank you for your interest!

We are aware of the paper, in fact we wrote a paper with one of the co-authors, [DYNOTEARS](https://arxiv.org/abs/2002.00498) that is also forthcoming at AISTAT20!
We did not know about this implementation, thank you for sharing.

That said, our current structure learning heavily relies on their original implementation.
There were several non-linear approaches published in the last year. Most of them add hyper parameters, making learning less practical for fast-paced projects. As many methods were developed simultaneously, we are waiting for a dominant approach to emerge. In addition, "linear" NOTEARS performs very well compared to previous structure learning methods even without non-linearity.

We are working on adding functionality for binary & categorical features. I can see how the pytorch implementation could help with a "logistic NOTEARS" using a softmax layer, also bias/intercepts become more important as well (but add complexity to the code).

We welcome contributions to `causalnex`!

Philip
